src/EditorFeatures/CSharpTest/MakeRefStruct/MakeRefStructTests.cs

```diff
@@ -0,0 +1,340 @@
+        {
+            var text = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+struct S
+{
+    Span<int>[||] m;
+}
+";
+
+            var expected = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+ref struct S
+{
+    Span<int> m;
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, parseOptions: s_parseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeRefStruct)]
+        [WorkItem(31831, "https://github.com/dotnet/roslyn/issues/31831")]
+        public async Task FieldInNestedClassInsideNotRefStruct()
+        {
+            var text = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+struct S
+{
+    class C
+    {
+        Span<int>[||] m;
+    }
+}
+";
+            await TestMissingInRegularAndScriptAsync(text, new TestParameters(s_parseOptions));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeRefStruct)]
+        [WorkItem(31831, "https://github.com/dotnet/roslyn/issues/31831")]
+        public async Task FieldStaticInRefStruct()
+        {
+            var text = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+ref struct S
+{
+    static Span<int>[||] m;
+}
+";
+            await TestMissingInRegularAndScriptAsync(text, new TestParameters(s_parseOptions));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeRefStruct)]
+        [WorkItem(31831, "https://github.com/dotnet/roslyn/issues/31831")]
+        public async Task FieldStaticInNotRefStruct()
+        {
+            var text = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+struct S
+{
+    static Span<int>[||] m;
+}
+";
+            var expected = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+ref struct S
+{
+    static Span<int> m;
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, parseOptions: s_parseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeRefStruct)]
+        [WorkItem(31831, "https://github.com/dotnet/roslyn/issues/31831")]
+        public async Task PropInNotRefStruct()
+        {
+            var text = @"
+using System;
+namespace System
+{
+    public readonly ref struct Span<T> 
+    {
+        unsafe public Span(void* pointer, int length) { }
+    }
+}
+
+struct S
+{
+    Span<int>[||] M { get; }
```

Comment:
note: this is an error scenario (code before&after won't compile). Maybe add a comment?

Reply:
I don't really understand what kind of comment should I write :confused:. It's obvious that if a codefix for a compiler error exists then the code is not compiled successfully.

Reply:
Aaah, you've talked about comments only for those cases where the code still can't be compiled after the fix applied, right?

Reply:
fixed